selfdrive/car/mazda/interface.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python3
+from cereal import car
+from selfdrive.config import Conversions as CV
+from selfdrive.controls.lib.drive_helpers import create_event, EventTypes as ET
+from selfdrive.controls.lib.vehicle_model import VehicleModel
+from selfdrive.car.mazda.values import CAR
+from selfdrive.car.mazda.carstate import CarState, get_powertrain_can_parser, get_cam_can_parser
+from selfdrive.car import STD_CARGO_KG, scale_rot_inertia, scale_tire_stiffness, gen_empty_fingerprint
+from selfdrive.car.interfaces import CarInterfaceBase
+
+class CanBus():
+  def __init__(self):
+    self.powertrain = 0
+    self.obstacle = 1
+    self.cam = 2
+
+ButtonType = car.CarState.ButtonEvent.Type
+
+class CarInterface(CarInterfaceBase):
+  def __init__(self, CP, CarController):
+    self.CP = CP
+
+    self.frame = 0
+    self.acc_active_prev = 0
+
+    # *** init the major players ***
+    canbus = CanBus()
+    self.CS = CarState(CP, canbus)
+    self.VM = VehicleModel(CP)
+    self.pt_cp = get_powertrain_can_parser(CP, canbus)
+    self.cam_cp = get_cam_can_parser(CP, canbus)
+
+    self.CC = None
+    if CarController is not None:
+      self.CC = CarController(canbus, CP.carFingerprint)
+
+  @staticmethod
+  def compute_gb(accel, speed):
+    return float(accel) / 4.0
+
+  @staticmethod
+  def get_params(candidate, fingerprint=gen_empty_fingerprint(), vin="", has_relay=False):
+    ret = car.CarParams.new_message()
+
+    ret.carName = "mazda"
+    ret.radarOffCan = True
+    ret.carFingerprint = candidate
+    ret.carVin = vin
+    ret.isPandaBlack = has_relay
+
+    ret.safetyModel = car.CarParams.SafetyModel.mazda
+
+    ret.enableCruise = True
+
+    # force openpilot to fake the stock camera, since car harness is not supported yet and old style
+    # giraffe (with switches) was never released
+    ret.enableCamera = True
+    
+    tire_stiffness_factor = 0.70   # not optimized yet
+
+    if candidate in [CAR.CX5]:
```

Comment:
Not needed

Reply:
What is not needed?

Reply:
I assume same as above,  u r referring to "if"

Reply:
sorry, my bad here too. The `if car...` here is fine.